packages/components/addon/components/hds/app-footer/status-link.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+
+export const STATUSES = {
+  OPERATIONAL: {
+    text: 'Operational',
+    iconName: 'check-circle-fill',
+    iconColor: 'var(--token-color-foreground-success)',
+  },
+  DEGRADED: {
+    text: 'Degraded',
+    iconName: 'alert-triangle',
+    iconColor: 'var(--token-color-foreground-warning)',
+  },
+  OUTAGE: {
+    text: 'Outage',
+    iconName: 'x-square-fill',
+    iconColor: 'var(--token-color-foreground-critical)',
+  },
+  MAINTENANCE: {
+    text: 'Maintenance',
+    iconName: 'info-fill',
+    iconColor: 'var(--token-color-foreground-highlight)',
+  },
+};
+
+export default class HdsAppFooterStatusLinkComponent extends Component {
+  /**
+   * @param ariaLabel
+   * @type {string}
+   * @description The name of the status which the StatusLink is being set to
+   */
+  get status() {
+    if (this.args.status) {
+      assert(
+        `@status for "Hds::AppFooter" must be one of the following: ${Object.keys(
+          STATUSES
+        ).join(', ')} received: ${this.args.status}`,
+        Object.keys(STATUSES).includes(this.args.status.toUpperCase())
+      );
+      return this.args.status.toUpperCase();
+    }
+    return undefined;
+  }
+
+  /**
+   * @param statusIcon
+   * @type {string}
+   * @description The name for the StatusLink icon
+   */
+  get statusIcon() {
+    if (this.status && !this.args.statusIcon) {
+      return STATUSES[this.status].iconName;
+    }
+    return this.args.statusIcon;
+  }
+
+  /**
+   * @param statusIconColor
+   * @type {string}
+   * @description The color for the StatusLink icon
+   */
+  get statusIconColor() {
+    if (this.status && !this.args.statusIconColor) {
+      return STATUSES[this.status].iconColor;
+    }
+    return this.args.statusIconColor;
+  }
+
+  /**
+   * @param text
+   * @type {string}
+   * @description The text content of the StatusLink
+   */
+  get text() {
+    if (this.status && !this.args.text) {
+      return STATUSES[this.status].text;
+    }
+    return this.args.text;
+  }
+
+  /**
+   * @param href
+   * @type {string}
+   * @description The href value of the StatusLink
+   */
+  get href() {
+    return this.args.href ?? 'https://status.hashicorp.com';
+  }
+
+  /**
+   * Get the class names to apply to the component.
+   * @method classNames
+   * @return {string} The "class" attribute to apply to the component.
+   */
+  get classNames() {
+    let classes = ['hds-app-footer__status-link'];
```

Comment:
I think this class is not used at all in the CSS

Reply:
It's a component. Your'e saying the component should not have a unique class name? Also there is still discussion around details of the visual design for this. The code has not been finalized.

Reply:
So I'm now using it in the CSS currently. Anyway going to resolve this and we can look at class names, etc. once the code is more finalized.